### PR TITLE
Create plugin-api crate and move most of core's logic into it

### PR DESCRIPTION
**Pull request by jeffcharles**
## Description of the change

Adds a library crate named `javy-plugin-api` with a lot `javy-core`'s logic. The idea is a Javy plugin developer should make their Javy plugin cdylib dependent on this `javy-plugin-api` crate and it will handle most of the exports for them and let them configure their runtime.

## Why am I making this change?

Part of #768. We don't want every plugin developer to have to re-implement most of the existing logic in `javy-core` themselves so this provides a crate they can depend on so they don't have to re-implement the existing logic.

I'm opting to leave the allocation function definitions in the `javy` library crate for this PR. I'll move the functions in the next PR.

## Checklist

- [x] I've updated the relevant CHANGELOG files if necessary. Changes to `javy-cli` and `javy-core` do not require updating CHANGELOG files.
- [x] I've updated the relevant crate versions if necessary. [Versioning policy for library crates](https://github.com/bytecodealliance/javy/blob/main/docs/contributing.md#versioning-for-library-crates)
- [x] I've updated documentation including crate documentation if necessary.

### CODE REVIEW

File: crates/core/Cargo.toml [hunk 1]
```diff
@@ -11,10 +11,9 @@ crate-type = ["cdylib"]
 
 [dependencies]
 anyhow = { workspace = true }
-javy = { workspace = true, features = ["export_alloc_fns", "json"] }
-once_cell = { workspace = true }
+javy-plugin-api = { workspace = true, features = ["json"] }
```
Comment (saulecabrera) [inline crates/core/Cargo.toml:14]: Is it intended to have this dependency at the workspace level? I think it's not shared across any other crates in the workspace, right?

File: crates/plugin-api/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
+//! A crate for creating Javy plugins
+//!
+//! Example usage:
+//! ```
```
Comment (saulecabrera) [inline crates/plugin-api/src/lib.rs:4]:
```suggestion
//! ```rust
```

Comment (saulecabrera) [reply crates/plugin-api/src/lib.rs:4]: To add syntax highlighting in LSP-powered highlighting.

**Review by saulecabrera — APPROVED** [review]
A couple of nits, but overall LGTM!

File: crates/core/Cargo.toml (hunk 1 above)
Comment (jeffcharles) [reply crates/core/Cargo.toml:14]: I can move it out of the workspace into `javy-core`. For some reason I thought the `javy` crate was only used by `javy-core` but I can see after looking that the fuzzer also uses it.

**Pull request merged**